src/app/shared/services/login/login.service.ts

```diff
@@ -138,30 +143,34 @@ export class LoginService {
       );
   }
 
-  private handleLoginErrors(e) {
+  /**
+   *  Handles non descriptive errors from iam-client-lib and cache server.
+   *  TODO: think about better implementation/approach.
+   */
+  private handleLoginErrors(e, navigateOnTimeout) {
     console.error(e);
     const loginError = LOGIN_ERRORS.filter(error => e.message.includes(error.key))[0];
     if (loginError?.type === 'swal') {
-      this.displayLoginErrorWithSwal(loginError.value);
+      const config = {
+        title: 'Wrong Network',
+        text: `${loginError.value}`,
+      };
+
+      this.openSwal(config, navigateOnTimeout);
+    } else if (e.message === 'Request failed with status code 401' || e.message === 'Request failed with status code 500') {
+      const config = {
+        title: 'Session Expired',
```

Comment:
Why does status code 500 indicate "Session Expired"? That doesn't make sense to me 🤔 .

Reply:
Well, i'm not sure why is that, but when i have old token (public key in local storage) and application is trying to login again with those old values, then i get 500. Removing publicKey with wallet provider from local storage and refreshing a page is fixing this. This looks for me that backend is sending 500 when gets an invalid token.

Reply:
Thanks for the explain. IMO, this is good place for a comment.
```suggestion
    } 
    // cache-server is returning 500 when old jwt (pubKey in local storage) is sent.
    // It seems therefore that cache-server is sending 500 when getting an invalid token.
    else if (e.message === 'Request failed with status code 401' || e.message === 'Request failed with status code 500') {
      const config = {
        title: 'Session Expired',
```

Reply:
Do you know if there is a bug logged for this in the cache-server backlog?

Reply:
I don't think so, i will test this one more time + i will look if there is a bug for this. If not then i will create it.